prow/apis/prowjobs/v1/types.go

```diff
@@ -302,6 +302,15 @@ type DecorationConfig struct {
 	// CookieFileSecret is the name of a kubernetes secret that contains
 	// a git http.cookiefile, which should be used during the cloning process.
 	CookiefileSecret string `json:"cookiefile_secret,omitempty"`
+	// OauthTokenSecret is a Kubernetes secret that contains the OAuth token,
+	// which is going to be used for fetching a private repository.
+	OauthTokenSecret *OauthTokenSecret `json:"oauth_token_secret,omitempty"`
```

Comment:
We should add a validation that prevents having both OAuth and SSH key configured

Reply:
Validation should be done in another PR and its better to merge it before this one. I am on it.